projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-wpcom-offline-subscription-service.php

```diff
@@ -37,19 +37,27 @@ public static function available() {
 	 * @throws \Exception Throws an exception when used outside of WPCOM.
 	 */
 	public function subscriber_can_receive_post_by_mail( $user_id, $post_id ) {
+		$is_blog_subscriber = true; // it is a subscriber as this is used in async job looping through subscribers...
+		$access_level       = get_post_meta( $post_id, '_jetpack_newsletter_access', true );
+		$allowed            = array(
+			Token_Subscription_Service::POST_ACCESS_LEVEL_SUBSCRIBERS,
+			Token_Subscription_Service::POST_ACCESS_LEVEL_EVERYBODY,
+		);
+
+		if ( in_array( $access_level, $allowed, true ) ) {
+			return true;
+		}
 
+		// All paid subscribers are users.
 		if ( 0 === $user_id || empty( $user_id ) ) {
```

Comment:
Mere email subscribers might be jetpack-only and therefore NOT have a $user_id?

Reply:
I am just wondering why we do not push that at the beginning of the function...

Reply:
It's been too long, but this was actually part of the issue.  If you look on WPCOM we already implemented this logic, Artur thought this PR would be smart to keep it in both places.  I can rebase if we want to merge or we can just drop it...

Reply:
I am struggling to get the full idea behind that PR right now...